fem/moonolith/pmortarassembler.cpp

```diff
@@ -32,1259 +32,1120 @@
+struct ParMortarAssembler::Impl {
+public:
+  MPI_Comm comm;
+  std::shared_ptr<ParFiniteElementSpace> source;
+  std::shared_ptr<ParFiniteElementSpace> destination;
+  std::vector<std::shared_ptr<MortarIntegrator>> integrators;
 
-   std::shared_ptr<IterativeSolver> solver;
+  std::shared_ptr<HypreParMatrix> coupling_matrix;
+  std::shared_ptr<HypreParMatrix> mass_matrix;
 
-   bool verbose{false};
+  std::shared_ptr<IterativeSolver> solver;
 
-   bool assemble_mass_and_coupling_together{true};
+  bool verbose{false};
 
-   void ensure_solver()
-   {
-      if(!solver) {
-         solver = std::make_shared<BiCGSTABSolver>(comm);
-         solver->SetMaxIter(20000);
-         solver->SetRelTol(1e-6);
-      }
-   }
-
-   bool is_vector_fe() const
-   {
-      bool is_vector_fe = false;
-      for (auto i_ptr : integrators)
-      {
-         if (i_ptr->is_vector_fe())
-         {
-            is_vector_fe = true;
-            break;
-         }
-      }
+  bool assemble_mass_and_coupling_together{true};
 
-      return is_vector_fe;
-   }
+  void ensure_solver() {
+    if (!solver) {
+      solver = std::make_shared<BiCGSTABSolver>(comm);
+      solver->SetMaxIter(20000);
+      solver->SetRelTol(1e-6);
+    }
+  }
 
-   BilinearFormIntegrator * new_mass_integrator() const
-   {
-      if (is_vector_fe())
-      {
-         return new VectorFEMassIntegrator();
-      }
-      else
-      {
-         return new MassIntegrator();
-      }
-   }
+  BilinearFormIntegrator *newBFormIntegrator() const {
+    assert(!integrators.empty());
+    return integrators[0]->newBFormIntegrator();
+  }
 };
 
 ParMortarAssembler::~ParMortarAssembler() = default;
 
-void ParMortarAssembler::SetSolver(const std::shared_ptr<IterativeSolver> &solver)
-{
-   impl_->solver = solver;
+void ParMortarAssembler::SetSolver(
+    const std::shared_ptr<IterativeSolver> &solver) {
+  impl_->solver = solver;
 }
 
-void ParMortarAssembler::SetAssembleMassAndCouplingTogether(const bool value)
-{
-   impl_->assemble_mass_and_coupling_together = value;
+void ParMortarAssembler::SetAssembleMassAndCouplingTogether(const bool value) {
+  impl_->assemble_mass_and_coupling_together = value;
 }
 
-void ParMortarAssembler::SetMaxSolverIterations(const int max_solver_iterations)
-{
-   impl_->ensure_solver();
+void ParMortarAssembler::SetMaxSolverIterations(
+    const int max_solver_iterations) {
+  impl_->ensure_solver();
 
-   impl_->solver->SetMaxIter(max_solver_iterations);
+  impl_->solver->SetMaxIter(max_solver_iterations);
 }
 
 void ParMortarAssembler::AddMortarIntegrator(
-   const std::shared_ptr<MortarIntegrator> &integrator)
-{
-   impl_->integrators.push_back(integrator);
+    const std::shared_ptr<MortarIntegrator> &integrator) {
+  impl_->integrators.push_back(integrator);
 }
 
-void ParMortarAssembler::SetVerbose(const bool verbose)
-{
-   impl_->verbose = verbose;
+void ParMortarAssembler::SetVerbose(const bool verbose) {
+  impl_->verbose = verbose;
```

Comment:
Have you changed the formatting of this file? Seems many of the changes are `3->2` indents and changing opening brackets. The mfem convention is what it used to be before the change.

Reply:
Applied ` astyle --options=$CODE_DIR/mfem/config/mfem.astylerc pmortarassembler.cpp`